### ISSUE TITLE
Once opened image is not updated

### ISSUE THREAD

**Issue by dmitriy-serdyuk** [jupyterlab/jupyterlab#3968]
Steps to recreate:
1) create an image
```python
plt.plot(numpy.array([1, 2, 3]))
plt.savefig('tst.png')
```
2) open it in a tab
![screen shot 2018-02-22 at 4 47 04 pm](https://user-images.githubusercontent.com/2872615/36565942-0c2e2536-17f0-11e8-8a16-276fe353568a.png)
3) change the image
```python
plt.plot(numpy.array([3, 2, 1]))
plt.savefig('tst.png')
```

Opened image does not update. Furthermore, closing and reopening the tab doesn't update the image either.

**Comment by ian-r-rose** [comment]
Hi @dmitriy-serdyuk, you are right that the opened image does not automatically re-read its contents from disk. However, when I close it and reopen it, the image is correctly updated.

Also, have you tried `Create New View for Output` from the Cell context menu? This may be a workaround for the behavior you are looking for.

**Comment by dmitriy-serdyuk** [comment]
Related: #541

**Comment by mimischi** [comment]
`Create New View for Output` works fine, but when I produce figures for publications, the preview in the output (enabled by `%matplotlib inline`) does not correspond to the image produces by `fig.savefig()`.

In this case, it would be awesome having the image opened in a separate tab reload automatically.

**Comment by flying-sheep** [comment]
Yes, I’d like to see that as well. Of course with all files, including e.g. PDFs.

I think Jupyter should monitor all opened files for modification (or replacement, some code might delete and recreate files) and replace the tab contents once it detects it.

**Comment by allenfrostline** [comment]
Hi following up on auto-refreshing readonly files. Is there any update on this feature since 2020? Seems https://github.com/jupyterlab/jupyterlab/issues/541 has been locked as resolved already

**Comment by paloha** [comment]
Even though there is an option under Files -> Reload File from Disk, it is very cumbersome. E.g. I am generating a pdf template and I want to see my changes as I iterate. I opened the pdf on the rigt half of the Jupyter Lab view in its own tab. Now when I regenerate the file I have to click on the pdf tab, then go to File -> Reload File from Disk, then go back to the coding tab to continue. Becomes annoying very quickly. 

Workaround that I found bearable:
* Right click the pdf file in the file browser and click "+ Open in new browser tab"
* Put the browser windows next to each oter
* Do your code changes
* Use keyboard shortcut to switch between current and last window (for me in Ubuntu <Alt + `>
* Hit F5 for refresh and you see the updated pdf
* Do <Alt + `> again to go back to coding

I know its not much, but the <Alt + `> + F5 speeds the process significantly.

**Comment by MMesch** [comment]
Instead of just solving this issue as in the fix in #17911, we may want to

- implement this for all files
- just check on refocus and ask the user whether they want to refresh to avoid polling regularly

**Comment by paloha** [comment]
@MMesch do you mean that on refocus - you want to check whether the file has changed, and if yes, only then ask the user if they want to refresh? Isn't the "check on refocus" and autorefresh simpler? Imagine you have many files open, your script updates all of them and then you have to click N-many confirmations.
